scala/BUILD

```diff
@@ -0,0 +1,20 @@
+load("//scala:scala_toolchain.bzl", "scala_toolchain")
+
+#call scala toolchain
+scala_toolchain(
+      name = 'scala_toolchain_impl',
+      scalacopts = [],
+      visibility = ["//visibility:public"]
+    )
+    
+toolchain_type(
```

Comment:
this is confusing to me. What is this doing? Is this declaring a new toolchain type?

Peeve: once I noticed this it really annoyed me: why does skylark use side-effecting functions with a `name` parameter rather than just let bindings:
```
scala_toolchain_type = toolchain_type(visibility = ["//visibility:public"])
```
ugh...

anyway.

Reply:
The syntax is a bit awkward. I agree.

Reply:
Yes, this declares a new toolchain type, which is currently just a label with no other data associated.